From 423416f58749de8e498afccc15d290a656abdf72 Mon Sep 17 00:00:00 2001
From: Gowrishankar Muthukrishnan <gmuthukr@redhat.com>
Date: Tue, 22 Oct 2019 10:59:14 +0530
Subject: [PATCH] lacp: report desync in ovs threads enabling slave

It is helpful in reporting main thread that is yet to enable bond slave,
but link state was brought up by lacp thread and capture this desync
between ovs threads for debugging.

Fixes: a8448cb170 ("lacp: Avoid packet drop on LACP bond after link up")
Signed-off-by: Gowrishankar Muthukrishnan <gmuthukr@redhat.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 ofproto/bond.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ofproto/bond.c b/ofproto/bond.c
index c5d5f2c03..3b148a244 100644
--- a/ofproto/bond.c
+++ b/ofproto/bond.c
@@ -817,6 +817,10 @@ bond_check_admissibility(struct bond *bond, const void *slave_,
          * When may_enable is TRUE, it means LACP is UP and waiting for the
          * main thread to run LACP state machine and enable the slave. */
         verdict = (slave->enabled || slave->may_enable) ? BV_ACCEPT : BV_DROP;
+        if (!slave->enabled && slave->may_enable) {
+            VLOG_DBG_RL(&rl, "bond %s: slave %s: main thread not yet enabled slave",
+                         bond->name, bond->active_slave->name);
+        }
         goto out;
     case LACP_CONFIGURED:
         if (!bond->lacp_fallback_ab) {
-- 
2.31.1

